salt/metalk8s/defaults.yaml

```diff
@@ -51,4 +53,36 @@ kubelet:
     authorization:
       mode: AlwaysAllow
 
+ca:
+  cert:
+    name: /etc/kubernetes/pki/ca.crt
+    signing_private_key: /etc/kubernetes/pki/ca.key
+    CN: kubenetes
+    keyusage: "critical digitalSignature, keyEncipherment, keyCertSign"
+    basicConstraints: "critical CA:true"
+    days_valid: 3650
+  signing_policies:
+    api:
+      - minions: '*'
+      - signing_private_key: /etc/kubernetes/pki/ca.key
+      - signing_cert: /etc/kubernetes/pki/ca.crt
+      - keyusage: "critical digitalSignature, keyEncipherment"
+      - extendedKeyUsage: "serverAuth"
+      - days_valid: 365
+
+kube_api:
+  cert:
+    name: /etc/kubernetes/pki/apiserver.crt
+    public_key: /etc/kubernetes/pki/apiserver.key
```

Comment:
Very strange to name this key public_key when we know that it will contain a private key. If I understand it is because you iterate over kube_api.cert to pass the args to x509.certificate_managed.
May be it's better if you rename this into `private_key` and just add an extra arg `public_key: kube_api.cert.public_key` with a little comment (not tested if it work)

Reply:
It's automatic btw when you give a private key as `public_key` arg that will take the `public_key` from this private key.
For me it's ok we just say we want to create a certificate for the public_key from this private_key.

Reply:
```
   public_key:
        The public key to be included in this certificate. This can be sourced
        from a public key, certificate, csr or private key. If a private key
        is used, the matching public key from the private key will be
        generated before any processing is done. This means you can request a
        certificate from a remote CA using a private key file as your
        public_key and only the public key will be sent across the network to
        the CA. If neither ``public_key`` or ``csr`` are specified, it will be
        assumed that this is a self-signed certificate, and the public key
        derived from ``signing_private_key`` will be used. Specify either
        ``public_key`` or ``csr``, not both. Because you can input a CSR as a
        public key or as a CSR, it is important to understand the difference.
        If you import a CSR as a public key, only the public key will be added
        to the certificate, subject or extension information in the CSR will
        be lost.
```

Reply:
Maybe a quick comment should help indeed, since this command is rather counter-intuitive.